Scripts/RemediationScripts/Remediate-EnableEncryptionInTransitForStorageAccounts.ps1

```diff
@@ -323,147 +413,269 @@ function Enable-StorageEncryptionInTransit
-                                                           @{N='StorageAccountName';E={$_.StorageAccountName}},
-                                                           @{N='EnableHttpsTrafficOnly';E={$_.EnableHttpsTrafficOnly}}                                                                  
+                $stgWithDisableHTTPS += $_|Select-Object  @{N='ResourceId';E={$_.Id}},
+                                                        @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                        @{N='StorageAccountName';E={$_.StorageAccountName}},
+                                                        @{N='EnableHttpsTrafficOnly';E={$_.EnableHttpsTrafficOnly}}                                                                  
             }
-        }                    
-                              
-        $totalStgWithEnableHTTPS = ($stgWithEnableHTTPS | Measure-Object).Count
-        $totalStgWithDisableHTTPS = ($stgWithDisableHTTPS | Measure-Object).Count
-    
-        Write-Host "Storage Account(s) with enabled 'secure transfer': [$($totalStgWithEnableHTTPS)]" -ForegroundColor $([Constants]::MessageType.Update)
-        Write-Host "Storage Account(s) with disabled 'secure transfer': [$($totalStgWithDisableHTTPS)]" -ForegroundColor $([Constants]::MessageType.Update)
+            Write-Host "Successfully fetched secure tranfer configuration of Storage Account: [$($_.StorageAccountName)]" -ForegroundColor $([Constants]::MessageType.Update)
+            Write-Host $([Constants]::SingleDashLine)
+        }
+        catch {
+            Write-Host "Error encountered while fetching secure transfer configuration of the Storage Account: [$($_.StorageAccountName)]." -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host "Skipping this Storage Account..." -ForegroundColor $([Constants]::MessageType.Warning)
+            $logResource = @{}
+            $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+            $logResource.Add("ResourceName",($_.ResourceName))
+            $logResource.Add("Reason","Error encountered while fetching secure transfer configuration of the Storage Account.")    
+            $logSkippedResources += $logResource
+            Write-Host $([Constants]::SingleDashLine)
+        }
+    }  
+    $totalStgWithEnableHTTPS = ($stgWithEnableHTTPS | Measure-Object).Count
+    $totalStgWithDisableHTTPS = ($stgWithDisableHTTPS | Measure-Object).Count
 
-        # Start remediation Storage Account(s) with 'secure transfer' enabled.
-        if ($totalStgWithDisableHTTPS -gt 0)
+    if ($totalStgWithDisableHTTPS -eq 0)
+    {
+        Write-Host "No Storage Account found with Secure Transfer disabled. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::DoubleDashLine)
+        if($AutoRemediation -and $totalStgWithEnableHTTPS -gt 0) 
```

Comment:
In case some resources got skipped due to error, do you think any warning or next steps will help user?

Reply:
Warning is given when any resource get skipped. Nothing on next steps

Reply:
What is expected from users to take care of warnings?

Reply:
There is no such instruction given earlier as well. So didn't includ anything.

Reply:
ok,